### PR TITLE
Surface Linux kernel 5.13.4 --> 5.16.11

### PR DESCRIPTION
**Pull request by kiana-S**
Potentially fixes issue #362

### CODE REVIEW

**Comment by kiana-S** [comment]
The build process seems to fail during patching. I don't see any reason why the 5.13 patches should work and the 5.16 patches don't, other than the 5.16 patches being incorrect in some way.

**Comment by Mic92** [comment]
Are you still working on this?

**Comment by kiana-S** [comment]
I've been trying on-and-off for a while now, but the patches are failing and I'm not sure why.

I'll look at it again this afternoon, and if I can't get the patches to work I'll probably close it.

**Comment by kiana-S** [comment]
After trying and failing to get the patches to work properly, I decided to reformat the entire linux derivation to use the pre-patched kernel at [linux-surface/kernel/v5.16-surface](https://github.com/linux-surface/kernel).

I have no idea if this actually fully works, but at least the kernel starts building.

**Comment by Mic92** [comment]
Ok. Let me know if the resulting kernel runs stable for you.

**Comment by kiana-S** [comment]
Complete success! The kernel runs stable, and #362 is fixed. I'll submit the pull request for review at this point.

File: microsoft/surface/repos.nix [hunk 1]
```diff
@@ -1,9 +1,7 @@
 { lib, pkgs, fetchFromGitHub, fetchurl }: {
-  linux-surface = fetchFromGitHub {
-    owner = "linux-surface";
-    repo = "linux-surface";
-    rev = "4130746c31606aebaccef58997b650beda9049f7";
-    sha256 = "0r74z0hj3rg8bn2arh2zj8p7awk7fi14gpyrs5h44lwxw4351xky";
+  linux-surface-kernel = fetchurl {
+    url = "https://github.com/linux-surface/kernel/archive/v5.16-surface.tar.gz";
```
Comment (Mic92) [inline microsoft/surface/repos.nix:3]: This is a branch, we should only reference commits or tags here. Otherwise the checksum will break.

Comment (Mic92) [reply microsoft/surface/repos.nix:3]: Also I don't quite understand the switch away from `fetchFromGitHub` here.

Comment (kiana-S) [reply microsoft/surface/repos.nix:3]: That's my bad; I misunderstood what `fetchFromGitHub` did and accidentally thought it didn't apply in this situation (I'm still a beginner, and nix isn't easy to pick up).
I'll fix it so it points to a specific revision too.

**Comment by Mic92** [comment]
merged in https://github.com/NixOS/nixos-hardware/pull/385

**Pull request closed without merging**